test/integration_test.py

```diff
@@ -404,6 +410,78 @@ def _get_entities_by_project(self, entity_type, project_short_name):
             params['search_after_uid'] = pagination['search_after_uid']
         return entities
 
+    def test_concurrent_portal_db_crud(self):
+        """
+        Use multithreading to simulate multiple users simultaneously modifying
+        the portals database.
+        """
+        if config.deployment_stage not in ('sandbox', 'nadove'):
+            log.info('Skipping portal CRUD test to avoid polluting database')
+            return
+
+        # Currently takes about 50 seconds and creates a 25 kb db file.
+        n_threads = 10
+        n_ops = 10
+        portal_service = PortalService()
+
+        # Assign to local variable outside of callback so we don't need to put
+        # the entire test inside of _crud call
+        portal_db = None
+
+        def get_db(db):
+            # `make format` is convinced this needs to be on two lines.
+            nonlocal \
```

Comment:
There must be a less convoluted way. `_crud` could just return what he callback returns.